context/amf_ue.go

```diff
@@ -32,6 +31,7 @@ import (
 	"github.com/omec-project/ngap/ngapType"
 	"github.com/omec-project/openapi/models"
 	"github.com/omec-project/util/fsm"
+	"github.com/omec-project/util/ueauth"
```

Comment:
Can you import UeauCommon so it avoids multiple code changes.

Reply:
The idea is to replace the single file (or few files) repos such as `UeauCommon` with the `util` repo (and use `util/ueauth` that has the same "code").

The only difference is that in `util/ueauth`, the `GetKDFValue` returns 2 values ([]byte, error) and the same function in `UeauCommon` returns only a single value ([]byte)

Reply:
> Can you import UeauCommon so it avoids multiple code changes.

FYI, this is the diff between the `UeauCommon` and the `util/ueauth`. and as I mentioned earlier, the only difference is that the `GetKDFValue` function returns 2 values instead of 1.

```diff
$ git diff UeauCommon/UeauCommon.go util/ueauth/ueauth.go
diff --git a/UeauCommon/UeauCommon.go b/util/ueauth/ueauth.go
index e74596d..20f632f 100644
--- a/UeauCommon/UeauCommon.go
+++ b/util/ueauth/ueauth.go
@@ -1,11 +1,15 @@
-package UeauCommon
+// Copyright 2019 Communication Service/Software Laboratory, National Chiao Tung University (free5gc.org)
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package ueauth

 import (
        "crypto/hmac"
        "crypto/sha256"
        "encoding/binary"
        "encoding/hex"
-       "log"
+       "fmt"
 )

 const (
@@ -20,7 +24,7 @@ const (
 )

 func KDFLen(input []byte) []byte {
-       var r = make([]byte, 2)
+       r := make([]byte, 2)
        binary.BigEndian.PutUint16(r, uint16(len(input)))
        return r
 }
@@ -28,15 +32,15 @@ func KDFLen(input []byte) []byte {
 // This function implements the KDF defined in TS.33220 cluase B.2.0.
 //
 // For P0-Pn, the ones that will be used directly as a string (e.g. "WLAN") should be type-casted by []byte(),
-// and the ones originially in hex (e.g. "bb52e91c747a") should be converted by using hex.DecodeString().
+// and the ones originally in hex (e.g. "bb52e91c747a") should be converted by using hex.DecodeString().
 //
 // For L0-Ln, use KDFLen() function to calculate them (e.g. KDFLen(P0)).
-func GetKDFValue(key []byte, FC string, param ...[]byte) []byte {
+func GetKDFValue(key []byte, FC string, param ...[]byte) ([]byte, error) {
        kdf := hmac.New(sha256.New, key)

        var S []byte
-       if STmp, err := hex.DecodeString(string(FC)); err != nil {
-               log.Printf("Hex decode failed: %+v", err)
+       if STmp, err := hex.DecodeString(FC); err != nil {
+               return nil, fmt.Errorf("GetKDFValue FC decode failed: %+v", err)
        } else {
                S = STmp
        }
@@ -46,8 +50,8 @@ func GetKDFValue(key []byte, FC string, param ...[]byte) []byte {
        }

        if _, err := kdf.Write(S); err != nil {
-               log.Printf("KDF write failed: %+v", err)
+               return nil, fmt.Errorf("GetKDFValue KDF write failed: %+v", err)
        }
        sum := kdf.Sum(nil)
-       return sum
+       return sum, nil
 }
```